File: Core Java/employee-app/src/com/techlabs/employee/Employee.java
package com.techlabs.employee;

import java.util.Calendar;
import java.util.Date;

public abstract class Employee {

	private String employeeName;
	private Date dateOfBirth;
	private double basicSalary;

	public Employee(String employeName, Date dateOfBirth, double basicSalary) {
		this.employeeName = employeName;
		this.dateOfBirth = dateOfBirth;
		this.basicSalary = basicSalary;
	}

	public String getEmployeeName() {
		return employeeName;
	}

	public Date getDateOfBirth() {
		return dateOfBirth;
	}

	public double getBasicSalary() {
		return basicSalary;
	}

	public float calculateAge() {
		float dob = 0;
		Calendar birthDate = Calendar.getInstance();
		birthDate.setTime(getDateOfBirth());
		Calendar today = Calendar.getInstance();
		float curYear = today.get(Calendar.YEAR);
		float birYear = birthDate.get(Calendar.YEAR);
		dob = curYear - birYear;
		float curMonth = today.get(Calendar.MONTH);

		float birMonth = birthDate.get(Calendar.MONTH);

		if (birMonth > curMonth) { // this year can't be counted!

			dob--;

		} else if (birMonth == curMonth) { // same month? check for day

			int curDay = today.get(Calendar.DAY_OF_MONTH);

			int dobDay = birthDate.get(Calendar.DAY_OF_MONTH);

			if (dobDay > curDay) { // this year can't be counted!

				dob--;

			}

		}

		return dob;
	}

	public abstract double calculateNetSalary();
}

File: DOTNET/C#/DLL/EmployeeSolution/EmployeeLibrary/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EmployeeLibrary
{
    public abstract class Employee
    {

        public String EmployeeName
        {
            get;
            set;
        }

        public DateTime DateOfBirth
        {
            get;
            set;
        }

        public double BasicSalary
        {
            get;
            set;
        }

        public double HouseRentAllowance
        {
            get
            {
               return  (BasicSalary * 50) / 100;
                
            }
            
        }

        public double DernessAllowance
        {
            get
            {
                return (BasicSalary * 40) / 100;
            }
        }

        public double Performance
        {
            get
            {
                return (BasicSalary * 20) / 100;
            }
        }

        public float CalculateAge()
        {
            float dob = 0;
            DateTime birthDate = DateOfBirth;
            DateTime todayDate = System.DateTime.Today;
            var todayYear = todayDate.Year;
            var birthYear = birthDate.Year;
            dob = todayYear-birthYear;

            var todayMonth = todayDate.Month;
            var birthMonth = birthDate.Month;

            if (birthMonth > todayMonth) {
                dob--;
            }
            else if(birthMonth == todayMonth)
            {
                var currentDay = todayDate.Day;
                var birthDay = birthDate.Day;

                if(birthDay > currentDay)
                {
                    dob--;
                }
            }
            return dob;
        }

        public abstract double CalculateSalary();
    }
}
